server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceParserHelper.java

```diff
@@ -91,9 +93,9 @@ private ValuesSourceParserHelper() {} // utility class, no instantiation
         if (timezoneAware) {
             objectParser.declareField(ValuesSourceAggregationBuilder::timeZone, p -> {
                 if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
-                    return DateTimeZone.forID(p.text());
+                    return ZoneId.of(p.text());
```

Comment:
Related to the version checks, should we support short IDs?  If not we should probably make a note in the breaking changes.

Reply:
we never use any short ids in tests, thus this has not happened uncovered yet. I thought this was not supported in joda...

Reply:
just to add to this everything returned by `DateTimeZone.getAvailableIDs()` is supported, so we are good here. There are more ids which are supported neither by joda nor by java time currently. Hope this clears it up.